samples/basic/fade_led/src/main.c

```diff
@@ -60,9 +67,16 @@ void main(void)
 	}
 
 	while (1) {
-		if (pwm_pin_set_usec(pwm_dev, PWM_CHANNEL,
-					PERIOD, pulse_width)) {
-			printk("pwm pin set fails\n");
+#if defined(CONFIG_SOC_ESP32)
+		/* ESP32 uses frequency instead of period. */
+		ret = pwm_pin_set_cycles(pwm_dev, PWM_CHANNEL,
```

Comment:
A quick look in `pwm.h` shows that `pwm_pin_set_usec()` is implemented in function of `pwm_pin_set_cycles()`.  It should work with ESP32.

Reply:
You're right, I was using this example before writing `pwm_get_cycles_per_sec()`, And without it, the example failed. I'll remove this commit.

But the fade can be done via hardware.